x-pack/plugins/ml/public/application/services/ml_api_service/index.ts

```diff
@@ -494,6 +497,30 @@ export function mlApiServicesProvider(httpService: HttpService) {
       });
     },
 
+    getVisualizerFieldHistograms({
+      indexPatternTitle,
+      query,
+      fields,
+      samplerShardSize,
+    }: {
+      indexPatternTitle: string;
+      query: any;
+      fields: FieldHistogramRequestConfig[];
+      samplerShardSize?: number;
+    }) {
+      const body = JSON.stringify({
+        query,
+        fields,
+        samplerShardSize,
+      });
+
+      return httpService.http<any>({
```

Comment:
I am repeatedly seeing this error the first time I try to view the histograms in the Transforms wizard.

The workflow I use is:
 - Open the ML plugin. Switch to the Data Frame Analytics page.
 - Starting creating a DFA job, and view the histograms for the source data.
 - Then switch to the Transform page in Stack Management.
 - Create a transform, and click the Histogram charts button 

![image](https://user-images.githubusercontent.com/7405507/87048640-e0be4100-c1f3-11ea-872e-9eada81d178c.png)

Reply:
Tested latest edit with https://github.com/elastic/kibana/pull/70976/commits/84203df4ebbf352f31a52f015778d7dc2c8f81e8  and this fixes the issue with `http` for me.

Reply:
@peteharverson It turned out having the endpoint only as part of the ML plugin doesn't guarantee that it's available when the transform plugin loads. To avoid a runtime dependency between the transforms and ML plugin (which would mean adding ML to transforms' `kibana.json`) I duplicated the API endpoint so each plugin has their own endpoint to get the histogram data. Both plugins still share the code to created the server side aggregations. The update is in 
a117e43, please have another look.